samples/cpp/train_HOG.cpp

```diff
@@ -403,23 +403,33 @@ void test_it( const Size & size )
 
 int main( int argc, char** argv )
 {
-    if( argc != 5 )
+    cv::CommandLineParser parser(argc, argv, "{help h|| show help message}"
+            "{pd||pos_dir}{p||pos.lst}{nd||neg_dir}{n||neg.lst}");
+    if (parser.has("help"))
     {
-        cout << "Wrong number of parameters." << endl
-            << "Usage: " << argv[0] << " pos_dir pos.lst neg_dir neg.lst" << endl
-            << "example: " << argv[0] << " /INRIA_dataset/ Train/pos.lst /INRIA_dataset/ Train/neg.lst" << endl;
-        exit( -1 );
+        parser.printMessage();
+        exit(0);
     }
     vector< Mat > pos_lst;
     vector< Mat > full_neg_lst;
     vector< Mat > neg_lst;
     vector< Mat > gradient_lst;
     vector< int > labels;
-
-    load_images( argv[1], argv[2], pos_lst );
+    string pos_dir = parser.get<string>("pd");
+    string pos = parser.get<string>("p");
+    string neg_dir = parser.get<string>("nd");
+    string neg = parser.get<string>("n");
+    if( !parser.check() )
```

Comment:
Please check whether any parameter is empty. All four are required for this program.